### ISSUE TITLE
Option is lowercased upon decorating function

### ISSUE THREAD

**Issue by ZenithDK** [pallets/click#1052]
pip list
Package        Version
-------------- ---------
...
click          6.7
...

```import click

@click.command()
@click.option("--someOption", help="Help text")
def test_fun(someOption):
    print("Test print!")

if __name__ == "__main__":
    test_fun()
```
This will give an error:
```
Traceback (most recent call last):
  File "AutoGenerateVoicePromptPackage.py", line 9, in <module>
    test_fun()
  File "C:\Python27\lib\site-packages\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "C:\Python27\lib\site-packages\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "C:\Python27\lib\site-packages\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Python27\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
TypeError: test_fun() got an unexpected keyword argument 'someoption'
```

As you can see, it expects the casing to be "someoption" - and not "someOption" as I specified it.
I'm running this on Windows 10 with Python 2.7.14 - could that be a factor?

**Comment by jcrotts** [comment]
I think this is expected behavior see this [line in core.py](https://github.com/pallets/click/blob/master/click/core.py#L1810)

What specifically are you trying to accomplish?

**Comment by ZenithDK** [comment]
This is wholly undocumented. I see the line of code and that is how I figured out it wasn't me that was causing the problem - but it is no documented.
Yes, all the examples use lowercase, but nowhere I can find does it say only lowercase is allowed (or that it is converted into lowercase).

**Comment by ZenithDK** [comment]
https://github.com/pallets/click/pull/1055

**Comment by jcrotts** [comment]
Resolved in #1055
